openshiftcontrolplane/v1/types.go

```diff
@@ -215,6 +215,11 @@ type ServiceAccountControllerConfig struct {
 type DockerPullSecretControllerConfig struct {
 	// registryURLs is a list of urls that the docker pull secrets should be valid for.
 	RegistryURLs []string `json:"registryURLs"`
+
+	// internalRegistryHostname is the hostname for the default internal image
+	// registry. The value must be in "hostname[:port]" format.  Docker pull secrets
+	// will be generated for this registry.
+	InternalRegistryHostname string `json:"internalRegistryHostname"`
```

Comment:
In usage does this need to be a slice to contain the different possible aliases:
 1. without suffix
 2. with suffix
 3. ip address

?

I'll merge it if you say it doesn't.  Just want to make sure you've considered it.

Reply:
no, historically internalRegistryHostname has always been a single value and the way it's used i think it can only be a single value, having it have multiple possible values would lead to weird inconsistencies.

i'm working on the unit test so don't merge yet.

Reply:
sorry I mixed up my PRs.  going to apply a lgtm to this.